CMakeLists.txt

```diff
@@ -391,6 +393,19 @@ set (Seastar_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 set (Seastar_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set (Seastar_GEN_BINARY_DIR ${Seastar_BINARY_DIR}/gen)
 
+include (TriStateOption)
+tri_state_option (${Seastar_SANITIZE}
+  DEFAULT_BUILD_TYPES "Debug" "Sanitize"
+  CONDITION sanitizers_enabled)
+
+if (sanitizers_enabled)
+  if ((NOT Seastar_SANITIZERS) OR (Seastar_SANITIZERS STREQUAL ""))
+    set (Seastar_SANITIZERS ${Seastar_DEFAULT_SANITIZERS})
```

Comment:
i'd suggest making `Seastar_SANITIZERS` a CMake option explicitly, like:
```cmake
set (Seastar_SANITIZERS
  "address;undefined_behavior"
  CACHE
  STRING
  "Sanitizers enabled when building Seastar")
```
and if we build with multi-config generator, this option applies to `Debug` and `Sanitize` modes, otherwise, this option applies the the current build.

what do you think?

Reply:
I agree about multi-config generator.

It doesn't make sense to use sanitizers in other modes because they (at least address and thread) won't work with seastar allocator. But you still can do it by passing `Seastar_SANITIZE=yes`.

I can force to use sanitizers in `./configure.py` if they are specified manually and `args.mode != 'all'`. What do you think?

P.S. `./configure.py` does not use multi-config generator

Reply:
Now I left the logic of sanitizers and build modes unchanged. Enabling sanitizers depends on `Seastar_SANITIZE` variable. By default it enables them only for Debug and Sanitize